src/gfx_d3d12.zig

```diff
@@ -50,7 +51,7 @@ const TextureHashMap = std.AutoHashMap(IdLocal, TextureHandle);
 // Material Pool
 const MaterialPool = Pool(16, 16, fd.PBRMaterial, struct { obj: fd.PBRMaterial });
 pub const MaterialHandle = MaterialPool.Handle;
-const MaterialHashMap = std.StringHashMap(MaterialHandle);
+const MaterialHashMap = std.AutoHashMap(IdLocal, MaterialHandle);
```

Comment:
Depending on how often this gets accessed I would in general not use IdLocal as keys, I would just use the hash. Might be that we should provide our own "wrapper" for it so that the hashmap doesn't hash the hash. But if it only gets accessed occasionally I think it's fine. Maybe a reasonable guideline is, if it's accessed multiple times per frame, use u64. If it's used less than once per second, IdLocal is fine. Inbetween.. use judgment :)

Reply:
For now all these hash maps are used when models (meshes, materials and textures) are loaded from disk and never during a frame. 
During a frame they're always referenced by handles.

Reply:
Ok! Sounds good. Potentially something we could look into when we have lots and lots of content that we need need to stream in live but this doesn't seem like it'd be much of a problem even then.